homeassistant/components/wittiot/config_flow.py

```diff
@@ -0,0 +1,72 @@
+"""Config flow for Wittiot integration."""
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+from wittiot import API
+from wittiot.errors import WittiotError
+
+from homeassistant import config_entries, exceptions
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_IP, CONNECTION_TYPE, DOMAIN, LOCAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Wittiot."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
```

Comment:
I suggest flipping the flow, 
```py
errors = {}
if user_input is not None:
    ... all checks go here
return self.async_show_form(errors=errors, ...)
```
This will reduce the amount of `async_show_form` from 3 to 1 and increases readability